test/Sema/exhaustive_switch.swift

```diff
@@ -1228,20 +1228,23 @@ enum SR11212Tests {
     switch u {
     case .upair((let x, let y)): return (x, y)
     // expected-warning@-1 {{a tuple pattern cannot match several associated values at once, implicitly tupling the associated values and trying to match that instead}}
+    // expected-note@-2 {{'upair' has 2 associated values, not a single associated value that is a tuple of 2 elements}}
```

Comment:
Overall, I agree with the sentiment, but I want to discuss the phrasing a bit:

```
// expected-warning@-1 {{implicit conversion to match tuple pattern against enum case with 2 associated values is deprecated}}
// expected-note@-7 {{'upair' declared here}}
// expected-note@-3 {{remove unnecessary parentheses to silence this warning}
```

I want to avoid using:

1. "implicit conversion" -- this makes it sound like we support(ed) implicit conversions, even though we do(did) not, the earlier behavior was a bug
2. "unnecessary" -- this might make a user think "the parentheses are not necessary but not necessarily wrong either... So why is this a warning? Adding redundant (unnecessary) parentheses shouldn't be a warning...?"
3. "deprecated" -- this might make a user think "wait, don't we have stability guarantees, how can they deprecate stuff like that?" or "if this was working just fine before, why'd they deprecate it without any good reason? Swift devs bad"

How about the following:

```
// expected-warning@-1 {{matching tuple pattern against enum case with 2 associated values is confusing}}
// expected-note@-7 {{'upair' declared here}}
// expected-note@-3 {{remove the parentheses to silence this warning}} {{16-17=}} {{29-30=}}
```

Reply:
I just want to point out that `implicit conversion` more of a compiler jargon and probably too technical to be useful. `confusing` should not be used as well, we need to make it clear that compiler does understand what is going on.

Reply:
The "confusing" was intended as a "confusing to other readers" but you're right, it could be interpreted as "confusing to the compiler" which would be bad. 😐 Any other suggestions? Saying it is "incorrect" is not appropriate, then one might wonder "why is this a warning instead of an error"...

Reply:
How about using `cannot_convert_single_tuple_into_multiple_arguments` e.g. `enum case 'upair' expects 2 separate arguments; remove extra parentheses to change tuple into separate arguments`?

Reply:
@xedin's wording sounds good to me.